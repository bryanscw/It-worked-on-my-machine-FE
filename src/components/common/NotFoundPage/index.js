import React from 'react'
import { Link } from 'react-router-dom'

export default function NotFoundPage() {
    return (
        <div className="container">
            <h1>404 Not Found</h1>
            <p>Sorry, the page you are looking for can't be found.</p>
            <Link className="btn btn-primary" to="/">Return to Home</Link>
        </div>
    )
}
