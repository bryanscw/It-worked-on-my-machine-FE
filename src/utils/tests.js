import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    return <Provider store={store}>{component}</Provider>;
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return <Router history={history}>{component}</Router>
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    return renderWithRedux(renderWithRouter(component, route), initialState);
}