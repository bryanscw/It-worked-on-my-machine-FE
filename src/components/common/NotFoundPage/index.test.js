import React from 'react'
import { cleanup } from '@testing-library/react'
import NotFoundPage from './index.js';
import { renderWithRouter } from '../../../utils/tests.js';

afterEach(cleanup);

it('should take a snapshot', () => {
    const { asFragment } = renderWithRouter(<NotFoundPage />);
    expect(asFragment()).toMatchSnapshot();
})