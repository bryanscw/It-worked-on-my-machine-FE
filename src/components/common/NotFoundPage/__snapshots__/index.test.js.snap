// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      404 Not Found
    </h1>
    <p>
      Sorry, the page you are looking for can't be found.
    </p>
    <a
      class="btn btn-primary"
      href="/"
    >
      Return to Home
    </a>
  </div>
</DocumentFragment>
`;
